### PR TITLE
[9.0] add missing trigger on is_web_subscription

### PR DESCRIPTION
**Pull request by robinkeunen**
only check 2nd commit

### CODE REVIEW

File: product_subscription_web_access/models/subscription.py [hunk 1]
```diff
@@ -37,8 +37,18 @@ class ProductSubscriptionRequest(models.Model):
...
+
+    @api.multi
+    def validate_request(self):
+        super(ProductSubscriptionRequest, self).validate_request()
+        self.websubscriber._compute_is_web_subscribed()
+
+    @api.multi
+    def confirm_paid(self):
+        super(ProductSubscriptionRequest, self).confirm_paid()
+        self.websubscriber._compute_is_web_subscribed()
```
Comment (robinkeunen) [inline product_subscription_web_access/models/subscription.py]: remove

**Review by houssine78 — APPROVED** [review]

File: product_subscription_web_access/models/res_partner.py [hunk 2]
```diff
@@ -43,12 +42,13 @@ def _compute_is_web_subscribed(self):
...
-                and r.is_web_subscription
-                and today
-                <= _pd(r.subscription_date) + timedelta(days=temp_access)
-            )
+            temp_access_limit = fields.Date.to_string(today - timedelta(days=temp_access))
+            open_requests = self.env['product.subscription.request'].search(
+                [('state', '=', 'sent'),
+                 ('is_web_subscription', '=', True),
+                 ('websubscriber', '=', partner.id),
+                 ('subscription_date', '>=', temp_access_limit),
```
Comment (robinkeunen) [inline product_subscription_web_access/models/res_partner.py]: check

File: product_subscription_web_access/models/subscription.py [hunk 3]
```diff
@@ -37,8 +37,17 @@ class ProductSubscriptionRequest(models.Model):
...
     )
+
+    @api.multi
+    def validate_request(self):
+        super(ProductSubscriptionRequest, self).validate_request()
+        self.websubscriber._compute_is_web_subscribed()
+
+    @api.multi
+    def confirm_paid(self):
+        super(ProductSubscriptionRequest, self).confirm_paid()
```
Comment (houssine78) [inline product_subscription_web_access/models/subscription.py]: @robinkeunen if you do nothing there then do not override.

File: product_subscription_web_access/models/res_partner.py (hunk 2 above)
Comment (houssine78) [reply product_subscription_web_access/models/res_partner.py]: 👍

File: product_subscription_web_access/models/subscription.py (hunk 3 above)
Comment (robinkeunen) [reply product_subscription_web_access/models/subscription.py]: j'avoue

**Comment by houssine78** [comment]
ok to merge !

**Pull request merged**